appservice/src/SiteWrapper.ts

```diff
@@ -362,12 +381,76 @@ export class SiteWrapper {
         await this.waitForDeploymentToComplete(kuduClient, outputChannel);
     }
 
+    private async getJsonRequest(url: string, requestOptions: WebResource, node: IAzureNode): Promise<Object[]> {
+        // Reference for GitHub REST routes
+        // https://developer.github.com/v3/
+        // Note: blank after user implies look up authorized user
+        try {
+            const gitHubResponse: string = await requestP.get(url, requestOptions);
+            return JSON.parse(gitHubResponse);
+        } catch (error) {
+            if (error.message.indexOf('401 - "{\"message\":\"Bad credentials\",\"documentation_url\":\"https://developer.github.com/v3\"}"') > -1) {
+                // the default error is just "Bad Credentials," which is an unhelpful error message
+                const tokenExpired: string = localize('tokenExpired', 'Azure\'s GitHub token has expired.  Reauthorize in the Portal under "Deployment options."');
+                const goToPortal: string = localize('goToPortal', 'Go to Portal');
+                const input: string | undefined = await vscode.window.showErrorMessage(tokenExpired, goToPortal);
+                if (input === goToPortal) {
+                    node.openInPortal();
+                    // this needs to be replaced with a custom error
```

Comment:
I don't like 'TODO's' as comments in code - we'll just forget about it. File an issue instead

Reply:
https://github.com/Microsoft/vscode-azuretools/issues/78